staging/src/k8s.io/kubectl/pkg/cmd/logs/logs_nodes.go

```diff
@@ -0,0 +1,583 @@
+	cmd.Flags().StringVar(&o.Role, "role", o.Role, "Set a label selector by node role to filter on.")
+	cmd.Flags().StringVarP(&o.Selector, "selector", "l", o.Selector, "Selector (label query) to filter on.")
+	cmd.Flags().BoolVar(&o.Raw, "raw", o.Raw, "Perform no transformation of the returned data.")
+	cmd.Flags().BoolVar(&o.Unify, "unify", o.Unify, "Interleave logs by sorting the output. Defaults on when viewing node journal logs.")
+}
+
+func (o *NodeLogsOptions) ToNodeLogOptions() (*nodev1alpha1.NodeLogOptions, error) {
+	logOptions := &nodev1alpha1.NodeLogOptions{
+		Boot:                         o.Boot,
+		BootChanged:                  o.BootChanged,
+		Path:                         o.Path,
+		InsecureSkipTLSVerifyBackend: o.InsecureSkipTLSVerifyBackend,
+	}
+
+	if len(o.Services) > 0 {
+		logOptions.Services = o.Services
+	}
+
+	if len(o.SinceTime) > 0 {
+		t, err := util.ParseRFC3339(o.SinceTime, metav1.Now)
+		if err != nil {
+			return nil, err
+		}
+
+		logOptions.SinceTime = &t
+	}
+
+	if len(o.UntilTime) > 0 {
+		t, err := util.ParseRFC3339(o.UntilTime, metav1.Now)
+		if err != nil {
+			return nil, err
+		}
+
+		logOptions.UntilTime = &t
+	}
+
+	if len(o.Output) > 0 {
+		logOptions.Format = o.Output
+	}
+
+	if len(o.Pattern) > 0 {
+		logOptions.Pattern = o.Pattern
+	}
+
+	if len(o.Selector) > 0 && !o.TailSpecified {
+		logOptions.TailLines = &nodeSelectorTail
+	} else if o.TailLines != -1 {
+		logOptions.TailLines = &o.TailLines
+	}
+
+	return logOptions, nil
+}
+
+func (o *NodeLogsOptions) Complete(f cmdutil.Factory, cmd *cobra.Command, args []string) error {
+	if !cmd.Flags().Lookup("unify").Changed {
+		o.Unify = o.Path == "journal"
+	}
+
+	o.Resources = args
+	o.BootChanged = cmd.Flag("boot").Changed
+	o.Raw = o.Raw || o.Path == "journal"
+
+	var err error
+	o.Options, err = o.ToNodeLogOptions()
+	if err != nil {
+		return err
+	}
+
+	o.RESTClientGetter = f
+	o.LogsForObject = polymorphichelpers.LogsForObjectFn
+
+	if o.Object == nil {
+		builder := f.NewBuilder().
+			WithScheme(scheme.Scheme, scheme.Scheme.PrioritizedVersionsAllGroups()...).
+			SingleResourceType()
+		if len(o.Resources) > 0 {
+			builder.ResourceNames("nodes", o.Resources...)
+		}
+		if len(o.Role) > 0 {
+			req, err := labels.NewRequirement(fmt.Sprintf("node-role.kubernetes.io/%s", o.Role), selection.Exists, nil)
+			if err != nil {
+				return fmt.Errorf("invalid --role: %v", err)
+			}
+			o.Selector = req.String()
+		}
+		if len(o.Selector) > 0 {
+			builder.ResourceTypes("nodes").LabelSelectorParam(o.Selector)
+		}
+		result := builder.Do()
+
+		o.Prefix = o.Prefix || !result.TargetsSingleItems()
+
+		infos, err := result.Infos()
+		if err != nil {
+			return err
+		}
+
+		if o.Selector == "" && len(infos) != 1 {
+			return fmt.Errorf("expected a node resource")
+		}
+
+		o.Object = infos[0].Object
+		if o.Selector != "" && len(o.Object.(*corev1.NodeList).Items) == 0 {
+			fmt.Fprint(o.ErrOut, "No node resources found.\n")
```

Comment:
Consider using `fmt.Fprintln()`.

```suggestion
			fmt.Fprintln(o.ErrOut, "No node resources found.")
```

Reply:
thanks, updated